### ISSUE TITLE
Fail to read from feature store cache prior to initialization

### ISSUE THREAD

**Issue by mmadson** [launchdarkly/java-client#107]
When a connection issue to launch darkly's servers is simulated for a new uninitialized LDClient and a persistent feature store (redis) is supplied, the expectation is that the feature store cache will be used to evaluate the flags, however in my tests it appears that prior to initialization the feature store cache is not consulted and the hardcoded default value is returned instead. This greatly reduces the effectiveness of a persistent feature store in the event that launch darkly's servers are down for an extended period of time. In such a scenario we would not be able to restart our servers because doing so would put our LDClients in an uninitialized state and our persistent cache would be useless. This also seems to contradict the documentation:

https://support.launchdarkly.com/hc/en-us/articles/115002374068-Do-you-support-Redis-caching-

which states:

```
Our SDKs can be configured to use Redis as a persistent store, so even if an initial connection to LaunchDarkly fails, the last flag configurations will be used instead of the fallbacks. 
```

I believe the source of the problem is in this bit of code from LDClient.class:236:

```
            if (!this.initialized()) {
                logger.warn("Evaluation called before Client has been initialized for feature flag " + featureKey + "; returning default value");
                this.sendFlagRequestEvent(featureKey, user, defaultValue, defaultValue, (Integer)null);
                return defaultValue;
            } else {
```

As you can see, if an initial connection is not established; the featureStore is never consulted.

**Comment by brooswit** [comment]
Hi, thanks for pointing this out. We discussed it and we agree with you. We have added this to our backlog and we will let you know when we have a solution.

**Comment by pkaeding** [comment]
Hi @mmadson!  I apologize for not updating this issue sooner. This should be fixed as of the [2.4.0](https://github.com/launchdarkly/java-client/releases/tag/2.4.0) release.

Please let me know if you run into any other issues.
